### ISSUE TITLE
handler not working [my mistake - it is working!]

### ISSUE THREAD

**Issue by paulovelho** [milica/angular-selectbox#1]
Hey, milica!
Can you create an example in the demo with a handler?

The function-call when an option is selected doesn't seem to be working here.

Cheers!

**Comment by paulovelho** [comment]
my bad. solved it... =X

**Comment by milica** [comment]
Hey Paulo

I'll check it anyway and make the example how to use it.

**Comment by paulovelho** [comment]
milica, 

basically, the main problem I was having was that the handler function was being called before the selected index were updated. So, inside the function, the selected value was still wrong.

I solved it moving the "$scope.handler()" call from the "selectItem" function to the "index" watch, like this:

```
        /* watch if selected index has been changed */
        $scope.$watch('index', function(n, o) {
            if (n !== o) {
                parseSelected();
                $scope.handler();
            }
        });
```

I will create a fork from your awesome project with this simple change anyway, ok?

Cheers!
